portfolio/src/main/java/com/google/sps/data/BlogHashMap.java

```diff
@@ -0,0 +1,175 @@
+          isTagSupported = true;
+          if (map.containsKey(message.getTag())){
+            map.get(message.getTag()).addLast(message);
+          } else {
+            // No such mapping with specified tag,
+            // Create first mapping with specified tag.
+            LinkedList<BlogMessage> firstValue = new LinkedList<BlogMessage>();
+            firstValue.addFirst(message); 
+            map.put(message.getTag(), firstValue);
+          }
+        } 
+      }
+      if (!isTagSupported) {
+        System.out.println("Tag not supported. Tag: " + message.getTag());
+        System.out.println("For BlogMessage@ " + message);
+      }
+    }
+
+  }
+
+  public static int getSize(Map<String, LinkedList<BlogMessage>> map) {
+    return map.size();
+  }
+
+  // Takes map, and
+  // Returns a linkedList of all values of BlogMessage type in the map. 
+  public static LinkedList<BlogMessage> getMessages(Map<String, LinkedList<BlogMessage>> map) {
+    System.out.println("Fetching all [LinkedList] Values in map...");
+
+    Set<String> keys = map.keySet();
+    LinkedList<BlogMessage> values = new LinkedList<BlogMessage>();
+    for (String key : keys) {
+      values.addAll(map.get(key));
+    }
+    return values;
+  }
+
+  // Returns a requested amount of all BlogMessages in map
+  public static LinkedList<BlogMessage> getMessages(Map<String, LinkedList<BlogMessage>> map, int limit) {
+    if (limit < 0) {
+      System.out.println("Cannot load this amount.");
+      return new LinkedList<>();
+    }
+
+    LinkedList<BlogMessage> values = getMessages(map);
+    if (limit > values.size()) {
+      limit = values.size();
+    } 
+
+    System.out.println("Fetching requested amount of [LinkedList] Values for all tags...");
+
+    Iterator<BlogMessage> iterateList = values.iterator();
+    LinkedList<BlogMessage> limitedValues = new LinkedList<BlogMessage>();
+    int i = 0;
+    while (iterateList.hasNext() && i < limit){
+      limitedValues.addLast(iterateList.next());
+      i++;
+    }
+    return limitedValues;
+  }
+
+  // Takes a tag, and
+  // Returns a linkedList of all values of BlogMessage type for the specified tag.
+  public static LinkedList<BlogMessage> getMessages(String tag, Map<String, LinkedList<BlogMessage>> map) {
+    System.out.println("Fetching all [LinkedList] Value(s) for specified tag...");
+
+    LinkedList<BlogMessage> values = new LinkedList<BlogMessage>();
+    if (map.containsKey(tag)) {
+      values.addAll(map.get(tag));
+    } else {
+      System.out.println("Tag not in map. Tag : " + tag);
+    }    
+    return values;
+  }
+
+  // Returns a requested amount of all BlogMessages for a specified tag.
+  public static LinkedList<BlogMessage>getMessages(String tag, Map<String, LinkedList<BlogMessage>> map, int limit) {
+    if (limit < 0) {
+      System.out.println("Cannot load this amount.");
+      return new LinkedList<>();
+    }
+
+    LinkedList<BlogMessage> values = getMessages(tag, map);
```

Comment:
Same here. We will be doing extra work that will be abandoned in the end. Please try to improve it.